ods_ci/utils/scripts/openshift/openid.py

```diff
@@ -0,0 +1,448 @@
+
+2. Delete a dynamic client
+export CLIENT_REGISTRATION_TOKEN=<CLIENT_REGISTRATION_TOKEN>
+#python3 ods_ci/utils/scripts/openshift/openid.py delete-client --deletion-endpoint https://myAuthServer.com/deletionEndpoint --client-name <client-name>
+
+3. Add OpenID identity provider (it assumes you are already logged in to the cluster)
+export CLIENT_ID=<CLIENT_ID>
+export CLIENT_SECRET=<CLIENT_SECRET>
+#python3 ods_ci/utils/scripts/openshift/openid.py add-openid-idp --idp-name openid --issuer-url https://myAuthServer.com --ocp-secret-name openid-secret
+
+4. Delete OpenID identity provider (it assumes you are already logged in to the cluster)
+#python3 ods_ci/utils/scripts/openshift/openid.py delete-openid-idp --idp-name openid
+
+5. Update redirect URIs for a dynamic client
+export CLIENT_REGISTRATION_TOKEN=<CLIENT_REGISTRATION_TOKEN>
+#python3 ods_ci/utils/scripts/openshift/openid.py update-redirect-uris --operation <add/remove> --update-endpoint https://myAuthServer.com/updateEndpoint --client-name <client-name> --redirect-uri <redirect-uri1> --redirect-uri <redirect-uri2> --jenkins-props-file <jenkins-props-file>
+"""
+
+from typing import Literal
+import click
+import base64
+from ods_ci.utils.scripts.logger import log
+import json
+import requests
+from string import Template
+from ods_ci.utils.scripts.util import execute_command
+import os
+
+REGISTRATION_BODY_TEMPLATE = Template("""
+    {
+        "application_type": "web",
+        "redirect_uris": [
+           ${redirect_uris}
+        ],
+        "client_name": "${client_name}",
+        "contacts": [
+            ${contact_emails}
+        ]
+    }
+""")
+
+OCP_DEFAULT_SECRET_NAME = "openid-idp-client-secret"
+CLIENT_PROPERTIES_FILE_DEFAULT = "client.properties"
+
+class OpenIdOps:
+    """Class for OpenID identity provider operations"""
+
+    def __init__(self):
+        """Initialize OpenIdOps"""
+        self.token = ""
+        self.jenkins_props_file = CLIENT_PROPERTIES_FILE_DEFAULT
+        self.client_registration_token = ""
+        self.idp_name = ""
+        self.client_id = ""
+        self.client_name = ""
+        self.client_secret = ""
+        self.registration_endpoint = ""
+        self.ocp_secret_name = OCP_DEFAULT_SECRET_NAME
+
+    def _write_jenkins_properties(self):
+        """Write client details to a properties file for Jenkins to read"""
+        if self.jenkins_props_file:
+            try:
+                with open(self.jenkins_props_file, "w") as f:
+                    f.write(f"CLIENT_NAME={self.client_name}\n")
+                    f.write(f"CLIENT_ID={self.client_id}\n")
+                    f.write(f"CLIENT_SECRET={self.client_secret}\n")
+                    f.write(f"CLIENT_REGISTRATION_TOKEN={self.client_registration_token}\n")
```

Comment:
Maybe you can do it in one `f.write` command, like:
```suggestion
                    f.write(
                        f"CLIENT_NAME={self.client_name}\n"
                        f"CLIENT_ID={self.client_id}\n"
                        f"CLIENT_SECRET={self.client_secret}\n"
                        f"CLIENT_REGISTRATION_TOKEN={self.client_registration_token}\n"
                    )
```

Reply:
yeah it makes sense. However I believe we wouldn't notice any difference due to the tiny size of what this is writing